src/main/java/Utils/Appointment.java

```diff
@@ -0,0 +1,239 @@
+
+    public int getAppointmentID() {
+        int count = 0;
+        try {
+            Class.forName("org.apache.derby.jdbc.ClientDriver");
+            String sql = "SELECT count(*) FROM APPOINTMENTS";
+            Connection connection = DriverManager.getConnection("jdbc:derby://localhost:1527/Appointments", username, password);
+            Statement stmt = connection.createStatement();
+            ResultSet rs = stmt.executeQuery(sql);
+            rs.next();
+            count = rs.getInt(1);
+        } catch (ClassNotFoundException | SQLException e) {
+            System.out.println(e);
+        }
+        return count + 1;
+    }
+
+    public ArrayList<Event> sortAppointmentsByDate(String appointmentDate, ArrayList<Event> event) {
+        ArrayList<Event> e = new ArrayList<>();
+
+        String[] temp = appointmentDate.split("-");
+
+        event.stream().map(new Function<Event, Event>() {
+            @Override
+            public Event apply(Event event1) {
+                System.out.println(event1.getDate());
+                return event1;
+            }
+        }).filter((event1) -> (event1.getDate().equals(appointmentDate))).forEachOrdered((event1) -> {
+            e.add(event1);
+        });
+        return e;
+    }
+
+    public int getDoctorIDbyName(String name) {
+        int docID = doctors.get(name);
+        return docID;
+    }
+
+    public ArrayList<String> getGeneratedSlots(int apptmLength, int doctorID, String date) {
+        ArrayList<Event> currEvents = new ArrayList<>();
+        ArrayList<String> intervals = new ArrayList<>();
+        String[] hours = new String[2];
+        boolean addSlot = false;
+
+        currEvents = retrieveEventsbyDoctorID(doctorID);
+        currEvents = sortAppointmentsByDate(date, currEvents);
+
+        try {
+            Calendar cal = Calendar.getInstance();
+            Calendar cal2 = Calendar.getInstance();
+            cal.setTime(timeFormat.parse(dayStart));
+            cal2.setTime(timeFormat.parse(dayEnd));
+            Time u = new java.sql.Time(cal.getTimeInMillis());
+            hours[0] = timeFormat.format(u);
+
+            while (cal.getTime().before(cal2.getTime())) {
+                cal.add(Calendar.MINUTE, apptmLength);
+                Time startTime = new java.sql.Time(cal.getTimeInMillis());
+                String x = timeFormat.format(startTime);
+                hours[1] = x;
+                addSlot = compareSelectedTime(hours, currEvents);
+                if (addSlot == true) {
+                    String hoursRange = hours[0] + '-' + hours[1];
+                    intervals.add(hoursRange);
+                }
+                String temp = hours[1];
+                hours = new String[2];
+                hours[0] = temp;
+
+            }
+        } catch (ParseException e) {
+            System.out.println(e);
+        }
+        return intervals;
+    }
+
+    public boolean compareSelectedTime(String[] time, ArrayList<Event> currEvents) {
```

Comment:
No change needed here but - When I see a boolean function personally I prefer to see a name like `IsAvailableSlot` or something like that. (I'm not exactly sure what the function does so you'd be able to give it a better name.)

That way when it is called in an if statement it is immediately clear what it is doing.
`addSlot = IsAvailableSlot(hours, currEvents);`